persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java

```diff
@@ -52,46 +60,67 @@ public class Init implements BLauncherCmd {
+        Path databaseConfigPath = Paths.get(this.sourcePath, KEYWORD_MODULES, KEYWORD_CLIENTS,
+                DATABASE_CONFIGURATION_BAL);
         if (helpFlag) {
             String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(COMMAND_IDENTIFIER);
             errStream.println(commandUsageInfo);
             return;
         }
+        Generate persistCmd;
+        try {
+            Class<?> persistClass = Class.forName("io.ballerina.persist.cmd.Generate");
+            persistCmd = (Generate) persistClass.getDeclaredConstructor().newInstance();
+            persistCmd.setSourcePath(Paths.get(sourcePath).toAbsolutePath().toString());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
         try  {
             balProject = ProjectLoader.loadProject(Paths.get(sourcePath));
-            name = balProject.currentPackage().descriptor().org().value() + "." + balProject.currentPackage()
-                    .descriptor().name().value() + "." + "clients";
+            configName = balProject.currentPackage().descriptor().name().value() + "." + KEYWORD_CLIENTS;
+            persistName = balProject.currentPackage().descriptor().name().value() + "." + DATABASE;
         } catch (ProjectException e) {
             errStream.println("Not a Ballerina project (or any parent up to mount point)\n" +
                     "You should run this command inside a Ballerina project. ");
             return;
         }
+        File persistToml = new File(persistTomlPath.toString());
+        File databaseConfig = new File(databaseConfigPath.toString());
+        if (persistToml.exists() || databaseConfig.exists()) {
+            errStream.println("`bal persist init` command can only be used for initializing the project");
+            return;
+        }
         if (!Files.exists(Paths.get(sourcePath, configPath))) {
             try {
+                createPersistToml(persistTomlPath);
+                outStream.println("created Persist.toml file with configurations.");
                 createConfigToml();
                 outStream.println("Created Config.toml file inside the Ballerina project");
-                updateBallerinaToml();
-                outStream.println("Updated Ballerina.toml file with mysql driver dependencies.");
+                persistCmd.execute();
             } catch (BalException e) {
                 errStream.println(e.getMessage());
             }
         } else {
             try {
+                createPersistToml(persistTomlPath);
+                outStream.println("created Persist.toml file with configurations.");
                 updateConfigToml();
                 outStream.println("Updated Config.toml file with default database configurations.");
-                updateBallerinaToml();
-                outStream.println("Updated Ballerina.toml file with mysql driver dependencies.");
+                persistCmd.execute();
             } catch (BalException e) {
```

Comment:
Why is there a different try catch, both seems to do the same thing

Reply:
updated in [8415ac4](https://github.com/ballerina-platform/persist-tools/pull/52/commits/8415ac40b0d7895fed58315bbc7624580d898835)